### PR TITLE
feat(anta.cli): Implement exec tech-support to replace script

### PR DESCRIPTION
**Pull request by titom73**
Implement new CLI entrypoint to collect tech-support in replacement of `scripts/collect-scheduled-show-tech.py`

Supported options:

```bash
❯ anta exec collect-tech-support --help
Usage: anta exec collect-tech-support [OPTIONS]

  Collect scheduled tech-support from eos devices.

Options:
  -o, --output PATH               Path for tests catalog
  -ssh, --ssh-port INTEGER        SSH port to use for connection
  -t, --tags TEXT                 List of tags using coma as separator:
                                  tag1,tag2,tag3
  --log-level, --log [debug|info|warning|critical]
                                  Logging level of the command
  --help                          Show this message and exit.
```

Execution example:

```bash
❯ anta exec collect-tech-support --tags spine
[09:09:54] INFO     Connecting to devices...                                    
           INFO     Created /mnt/flash/schedule/all_files.zip on device spine01 
           INFO     Connected (version 2.0, client OpenSSH_7.8)                 
           INFO     Authentication (keyboard-interactive) successful!           
[09:09:55] INFO     Deleted /mnt/flash/schedule/all_files.zip on spine01        
           INFO     Created /mnt/flash/schedule/all_files.zip on device spine02 
           INFO     Connected (version 2.0, client OpenSSH_7.8)                 
           INFO     Authentication (keyboard-interactive) successful!           
[09:09:56] INFO     Deleted /mnt/flash/schedule/all_files.zip on spine02        
           INFO     Done collecting scheduled show-tech                         
```

### CODE REVIEW

File: anta/cli/exec/utils.py [hunk 1]
```diff
@@ -113,3 +120,78 @@ async def collect_commands(inv: AntaInventory,  enable_pass: str, commands: Dict
...
+                    device.name, root_dir)
+                # Connect to the dpreevice using SSH
+                ssh = create_ssh_client(
+                    device.host, ssh_port, device.username, device.password
+                )
+                # Get the zipped file all_files.zip using SCP and save it locally
+                my_path = f"{output_dir[1]}/{CURRENT_DATE}_{device.name}.zip"
+                scp = SCPClient(ssh.get_transport())
+                scp.get(ZIP_FILE, local_path=my_path)
+                scp.close()
```
Comment (gmuloc) [inline anta/cli/exec/utils.py:183]: on doit pouvoir utiliser un context ici comme la classe implémente __enter__ et __exit__
https://github.com/jbardin/scp.py/blob/master/scp.py

```suggestion
                with SCPClient(ssh.get_transport()) as scp:
                    scp.get(ZIP_FILE, local_path=my_path)
```

File: anta/cli/exec/utils.py [hunk 2]
```diff
@@ -113,3 +120,78 @@ async def collect_commands(inv: AntaInventory,  enable_pass: str, commands: Dict
...
+                # Delete the created zip file on the device
+                await device.session.cli(command=f"bash timeout 30 rm {ZIP_FILE}")
+                logger.info(f"Deleted {ZIP_FILE} on {device.name}")
```
Comment (gmuloc) [inline anta/cli/exec/utils.py:186]: Maybe let's handle the failure of deletion separately for this? to be able to give meaningful message to the user?

File: anta/cli/exec/utils.py [hunk 3]
```diff
@@ -113,3 +120,78 @@ async def collect_commands(inv: AntaInventory,  enable_pass: str, commands: Dict
...
+                logger.info(f"Created {ZIP_FILE} on device {device.name}")
+                # Create directories
+                output_dir = device_directories_show_tech_support(
+                    device.name, root_dir)
+                # Connect to the dpreevice using SSH
+                ssh = create_ssh_client(
+                    device.host, ssh_port, device.username, device.password
+                )
+                # Get the zipped file all_files.zip using SCP and save it locally
+                my_path = f"{output_dir[1]}/{CURRENT_DATE}_{device.name}.zip"
```
Comment (gmuloc) [inline anta/cli/exec/utils.py:180]: If you adopt my suggestion above need to change the `my_path` creation (probably can change the name to something more meaningful)
```suggestion
                output_dir = device_directories_show_tech_support(
                    device.name, root_dir)
                my_path = f"{output_dir[1]}/{CURRENT_DATE}_{device.name}.zip"
                # Connect to the dpreevice using SSH
                ssh = create_ssh_client(
                    device.host, ssh_port, device.username, device.password
                )
                # Get the zipped file all_files.zip using SCP and save it locally
```

File: anta/cli/exec/utils.py [hunk 4]
```diff
@@ -113,3 +120,78 @@ async def collect_commands(inv: AntaInventory,  enable_pass: str, commands: Dict
...
+
+def create_ssh_client(
+    dev: str, port: int, username: str, password: str, banner_timeout: int = 60
+) -> paramiko.SSHClient:
+    """
+    return a connected ssh client
+    """
+    client = paramiko.SSHClient()
+    client.set_missing_host_key_policy(paramiko.AutoAddPolicy())
+    client.connect(
```
[the remark this answers is not in the document]
Comment (titom73) [reply anta/cli/exec/utils.py:146]: Adding exceptions management

File: anta/cli/exec/utils.py [hunk 5]
```diff
@@ -113,3 +120,78 @@ async def collect_commands(inv: AntaInventory,  enable_pass: str, commands: Dict
...
+    """
+    return a tuple containing the show_tech_directory and the device_directory
+    """
+    cwd = os.getcwd()
+    show_tech_directory = f"{cwd}/{root_dir}"
+    device_directory = f"{show_tech_directory}/{dev}"
+    for directory in [show_tech_directory, device_directory]:
+        if not os.path.exists(directory):
+            os.makedirs(directory)
+    return show_tech_directory, device_directory
```
[the remark this answers is not in the document]
Comment (titom73) [reply anta/cli/exec/utils.py:135]: Agree. Change in progress

File: anta/cli/exec/utils.py [hunk 6]
```diff
@@ -113,3 +121,66 @@ async def collect_commands(inv: AntaInventory,  enable_pass: str, commands: Dict
...
+                logger.info(f"Created {EOS_TECH_SUPPORT_ARCHIVE_ZIP} on device {device.name}")
+
+            except EapiCommandError as e:
+                logger.error(f"Unable to create tech-support archove on {device.name}: {e.errmsg}")
+
+            # Create directories
+            tech_support_root_local_path = device_directories_show_tech_support(
+                device.name, root_dir)
+
+            # Connect to the dpreevice using SSH
```
Comment (gmuloc) [inline anta/cli/exec/utils.py:161]:
```suggestion
            # Connect to the device using SSH
```

it came back!

File: anta/cli/exec/utils.py [hunk 7]
```diff
@@ -113,3 +121,66 @@ async def collect_commands(inv: AntaInventory,  enable_pass: str, commands: Dict
...
+            ssh = device.create_ssh_socket(ssh_port=22)
```
Comment (gmuloc) [inline anta/cli/exec/utils.py:162]:
```suggestion
            ssh = device.create_ssh_socket(ssh_port)
```

**Review by gmuloc — COMMENTED** [review]
two more stuff :) and then go

File: anta/cli/exec/utils.py [hunk 8]
```diff
@@ -9,13 +9,19 @@
...
 
 from aioeapi import EapiCommandError
+from scp import SCPClient
 
 from anta.inventory import AntaInventory
 from anta.inventory.models import InventoryDevice
 
+EOS_TECH_SUPPORT_ARCHIVE_ZIP = "/mnt/flash/schedule/all_files.zip"
+
+CURRENT_DATE = strftime("%d %b %Y %H:%M:%S", gmtime())
```
Comment (gmuloc) [inline anta/cli/exec/utils.py:23]: we can remove this now I think?

**Review by gmuloc — APPROVED** [review]
LGTM - looks great - one last comment to remove unused constant and we can merge! Thanks for this :)

**Pull request merged**